### PR TITLE
[Rust][CI] Move CI over to new Rust crates and try to fix flaky test.

### PR DESCRIPTION
**Pull request by jroesch**
This PR finishes the move to the new Rust bindings and tries to integrate them into existing TVM docs and CI.

### CODE REVIEW

**Comment by tqchen** [comment]
@jroesch can you comment out the flaky error first? since it is confirmed but was not fixed https://github.com/apache/incubator-tvm/issues/6009

File: rust/tvm-rt/src/to_boxed_fn.rs [hunk 1]
```diff
@@ -43,9 +43,8 @@ where
 {
```
Comment (mwillsey) [inline rust/tvm-rt/src/to_boxed_fn.rs:43]: The whole `ToBoxedFn` seems a little strange. A trait called `To` should have function that takes `Self`, not return `Self`. It seems to be implemented backwards.

Could you instead just have a generic, inherent method on `Function`?

Comment (mwillsey) [reply rust/tvm-rt/src/to_boxed_fn.rs:43]: Ah nvm, this is unrelated to this change.

**Comment by tqchen** [comment]
THanks @jroesch @mwillsey , this PR is now merged

**Pull request merged**